lib/ui/page/buy.dart

```diff
@@ -0,0 +1,437 @@
+  }) : super(key: key);
+
+  final List<AssetResult> supportedAssets;
+  final List<WyreFiat> fiatList;
+
+  final _debouncer = Debouncer(const Duration(milliseconds: 500));
+
+  @override
+  Widget build(BuildContext context) {
+    final asset = useState(supportedAssets[0]);
+    final fiat = useState(fiatList[0]);
+    final type = useState(WyrePayType.applePay);
+    final wyreQuote = useState<WyreQuote?>(null);
+    final lastQuoteByFiat = useState(true);
+
+    final fiatController = useTextEditingController();
+    final cryptoController = useTextEditingController();
+    final fiatFocusNode = useFocusNode(debugLabel: 'fiat input');
+    final cryptoFocusNode = useFocusNode(debugLabel: 'crypto input');
+
+    final country = useMemoized(getCountry);
+
+    void showAssetListBottomSheet() {
+      showMixinBottomSheet(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => AssetSelectionListWidget(
+          onTap: (AssetResult assetResult) {
+            context.replace(buyPath.toUri({'id': assetResult.assetId}));
+            asset.value = assetResult;
+          },
+          selectedAssetId: asset.value.assetId,
+          assetResultList: supportedAssets,
+        ),
+      );
+    }
+
+    void showFiatListBottomSheet() {
+      showMixinBottomSheet(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => FiatSelectionListWidget(
+          fiatList: fiatList,
+          selectedFiat: fiat.value,
+          onTap: (WyreFiat wyreFiat) {
+            fiat.value = wyreFiat;
+          },
+        ),
+      );
+    }
+
+    Future<WyreQuote> queryOrderQuote(bool byFiat) async {
+      final data = {
+        'sourceCurrency': fiat.value.name,
+        'destCurrency': asset.value.symbol,
+        'dest': 'ethereum:${asset.value.destination}',
+        'country': country,
+        'accountId': Env.wyreAccount,
+        'walletType': type.value.forQuote(),
+        'amountIncludeFees': 'true',
+      };
+      if (byFiat) {
+        data['sourceAmount'] = fiatController.text;
+      } else {
+        data['destAmount'] = cryptoController.text;
+      }
+      final quote =
+          await WyreClient.instance.api.getOrderReservationQuote(data);
+      return quote;
+    }
+
+    useEffect(() {
+      void updateAmount() {
+        _debouncer.run(() async {
+          final amount = double.tryParse(fiatController.text) ?? 0;
+          if (amount == 0) {
+            cryptoController.text = '';
+            wyreQuote.value = null;
+            return;
+          }
+
+          if (fiatFocusNode.hasFocus) {
+            final quote = await queryOrderQuote(true);
+            cryptoController.text = quote.destAmount.toString();
+            wyreQuote.value = quote;
+            lastQuoteByFiat.value = true;
+          }
+        });
+      }
+
+      fiatController.addListener(updateAmount);
+      return () {
+        fiatController.removeListener(updateAmount);
```

Comment:
does `_debouncer`'s _timer also be canceled in  here?

Reply:
removed _debouncer